src/main/java/com/cleanroommc/groovyscript/compat/mods/pyrotech/BrickSawmill.java

```diff
@@ -0,0 +1,146 @@
+                .output(output)
+                .register();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.ADDITION, example = @Example("'glowstone_to_dust', item('minecraft:glowstone'), item('pyrotech:sawmill_blade_stone'), item('minecraft:glowstone_dust'), 200, 0"))
+    public BrickSawmillRecipe add(String name, IIngredient input, IIngredient blade, ItemStack output, int duration, int woodChips) {
+        return recipeBuilder()
+                .duration(duration)
+                .woodChips(woodChips)
+                .name(name)
+                .input(input, blade)
+                .output(output)
+                .register();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.REMOVAL, example = @Example("item('minecraft:planks:1')"))
+    public void removeByInput(ItemStack input) {
+        if (GroovyLog.msg("Error removing refractory sawmill recipe")
+                .add(IngredientHelper.isEmpty(input), () -> "Input 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (BrickSawmillRecipe recipe : getRegistry()) {
+            if (recipe.getInput().test(input)) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @MethodDescription(type = MethodDescription.Type.REMOVAL, example = @Example("item('pyrotech:material:23')"))
+    public void removeByOutput(IIngredient output) {
+        if (GroovyLog.msg("Error removing refractory sawmill recipe")
+                .add(IngredientHelper.isEmpty(output), () -> "Output 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (BrickSawmillRecipe recipe : getRegistry()) {
+            if (output.test(recipe.getOutput())) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @Property(property = "input", comp = @Comp(gte = 1, lt = 3))
+    @Property(property = "output", comp = @Comp(eq = 1))
+    @Property(property = "name")
+    public static class RecipeBuilder extends AbstractRecipeBuilder<BrickSawmillRecipe> {
+
+        @Property(comp = @Comp(gte = 1))
+        private int duration;
+        @Property(comp = @Comp(gte = 0))
+        private int woodChips;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder duration(int time) {
+            this.duration = time;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder woodChips(int woodChips) {
+            this.woodChips = woodChips;
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Pyrotech Refractory Sawmill Recipe";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 1, 2, 1, 1);
+            msg.add(duration < 0, "duration must be a non negative integer, yet it was {}", duration);
```

Comment:
this allows a duration of `0`, which is annotated `@Property(comp = @Comp(gte = 1))`. one of these should be adjusted